nautobot/dcim/forms.py

```diff
@@ -4202,7 +4192,7 @@ class PowerFeedForm(NautobotModelForm):
     location = DynamicModelChoiceField(
         queryset=Location.objects.all(),
         required=False,
-        initial_params={"powerpanel": "$power_panel"},
+        initial_params={"powerpanels": "$power_panel"},
```

Comment:
Hm. Not `power_panels`?

Reply:
I think `initial_params` are referencing the actual parent model field in this case `powerpanels`. Unlike `query_params` which are using the filterset field. I got a FieldError when I changed it to `power_panels`.
```
# Set initial value based on prescribed child fields (if not already set)
        if not self.initial and self.initial_params:
            filter_kwargs = {}
            for kwarg, child_field in self.initial_params.items():
                value = form.initial.get(child_field.lstrip("$"))
                if value:
                    filter_kwargs[kwarg] = value
            if filter_kwargs:
                self.initial = self.queryset.filter(**filter_kwargs).first()
```

Reply:
We should change it back when the dcim related names and FK issue is closed tho https://github.com/nautobot/nautobot/issues/3066. Leaving a note here

Reply:
Got it. Thanks!